modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/resource/v1_0/NavigationMenuResourceImpl.java

```diff
@@ -440,6 +448,85 @@ private String _getName(
 		return unicodeProperties.getProperty("title");
 	}
 
+	private Object _getNavigationMenuItemSettings(
+		String type, UnicodeProperties unicodeProperties) {
+
+		if (Objects.equals(
+				type, SiteNavigationMenuItemTypeConstants.ASSET_VOCABULARY)) {
+
+			return new VocabularyNavigationMenuItemSettings() {
```

Comment:
Would it be cleaner/simpler to have the logic in its own DTO class? Like this https://github.com/liferay/liferay-portal/blob/ac0a965a86a2b89faa1a95275fff633b6feb025f/modules/apps/headless/headless-admin-site/headless-admin-site-impl/src/main/java/com/liferay/headless/admin/site/internal/dto/v1_0/converter/PageElementDTOConverter.java#L124